server/types.ts

```diff
@@ -29,4 +29,15 @@ export type DiscordAccount = {
 
 type GameMode = "taiko" | "catch"; // supported osu! gamemodes
 
+/**
+ * Maps gamemodes to their legacy ID
+ */
+export const legacyGamemodeId = Object.freeze({
```

Comment:
a function for this is also defined in util.ts, getGamemodeId -- we should probably consolidate it to one place

Reply:
Ahh didn't know that, d4c25040276da79ecc5d20a5ff4b2d3a4cbaa008 now uses a function defined in `utils.ts`

I also went ahead and added other known game modes to `GameMode`'s typedef and the `MODE_TO_ID` map. In both of these both `catch` and `fruits` are defined though, not sure if that's desirable or `GameMode` typedef